[EDU-3187] feat: add load balance a dns (#645)
aziontech/docs: src/content/docs/en/pages/secure-journey/intelligent-dns-advanced-configurations/load-balance-dns.mdx

description: Perform a DNS load balance by assigning different weights to records configured in Intelligent DNS.
meta_tags: dns, intelligent dns, load balance, records
namespace: docs_secure_load_balance_dns
menu_namespace: secureMenu
permalink: /documentation/products/guides/secure/load-balance-dns/
---

import ContributorList from '~/components/ContributorList.astro'

You can use a [Intelligent DNS zone](/en/documentation/products/secure/intelligent-dns/) to create more than one record and perform a DNS load balance. Assigning different weights to each record for a same zone helps distribute incoming network traffic, ensuring optimal resource utilization and preventing overload on a single server.

In this guide, you'll learn an example using a record type `A`, which accepts values of IPv4 address format.

---

## Via RTM

1. [Access RTM](/en/documentation/products/guides/how-to-access-rtm/) > **Intelligent DNS**.
2. From the list, choose the zone in which you want to add records. Or, [create a zone](/en/documentation/products/guides/secure/intelligent-dns-configure-main-settings/#creating-a-new-zone).
3. Select the **Records** tab.
4. Click the **Add Record** button.

> 5. In **Name**, provide the new record's name. Example: `load balance dns record`.

Author: ```suggestion
5. In **Name**, provide the new record's name. Example: `lbexample`.
```

6. In **Type**, select the type of record `A`.
7. In **Value**, add the items for the DNS response to the registered record in IPv4 address format. Example: `192.111.0.1`.
8. In **TTL (seconds)**, choose the time, in seconds, that a response can be cached on a resolver server. Maximum value: `2147483647`.
9. In **Policy**, select **Weighted**.
    - In **Weight**, specify the weight for the record, considering that you'll create other records for the same zone with different weights. Accepts values from `0` to `255`.
    - In **Description** (optional), you can add a short text that differentiates the records you'll create for the load balance. Accepts up to *45 characters*.
10. Click the **Save** button.

> 11. Repeat steps 4 to 10 as many times as you want, but changing the **Weight** field.

Author: ```suggestion
11. Repeat steps 4 to 10 as many times as you want, but changing the **Value** and **Weight** fields.
```

> 11. Repeat steps 4 to 10 as many times as you want, but changing the **Value** and **Weight** fields.
>     - Optionally, you can choose different values for the **Name** and **Description** fields.

Author: ```suggestion
11. Repeat steps 4 to 10 to create the other balanced DNS records needed while observing Intelligent DNS limits, and always using the same **Name** and setting the desired addresses in **Value** and weights for each record in **Weight**.
```

If you add 3 records, you can, for example, specify a weight of **50** for the first record, **20** for the second record, and **30** for the third record.

---

## Via API

1. Run the following `GET` request in your terminal, replacing `[TOKEN VALUE]` with your [personal token](/en/documentation/products/guides/personal-tokens/) to retrieve your `<hosted_zone_id>`:

```bash
curl --location 'https://api.azionapi.net/intelligent_dns' \
--header 'Authorization: Token [TOKEN VALUE]' \
--header 'Accept: application/json; version=3'
```

2. You'll receive a response with all your existing edge firewalls. Copy the value of the `<id>` that you want to use.
3. Run the following `POST` request, replacing `[TOKEN VALUE]` with your [personal token](/en/documentation/products/guides/personal-tokens/) and the `<hosted_zone_id>` value you copied:

```bash
curl --location 'https://api.azionapi.net/intelligent_dns/<hosted_zone_id>/records' \
--header 'Content-Type: application/json' \
--header 'Authorization: Token [TOKEN VALUE]' \
--header 'Accept: application/json; version=3' \
--data-raw '{
    "record_type": "A",

>     "entry": "load balance dns record",

Author: ```suggestion
    "entry": "lbexample",
```

    "answers_list": [
        "192.111.0.1",
        "192.111.0.2"
    ],
    "ttl": 20,
    "policy": "weighted",

>     "weight": 50

Author: ```suggestion
    "weight": 50,
```

    "description": "policy weight 50"
}'
```

4. You'll receive a response similar to this:

```json
{
  "results": {
    "answers_list": [
      "192.111.0.1",
      "192.111.0.2"
    ],
    "zone_id": 1234,
    "record_type": "A",
    "ttl": 20,
    "policy": "weighted",
    "weight": 50,
    "entry": "load balance dns record",
    "id": 12345,
    "description": "policy weight 50"
  },
  "schema_version": 3
}
```

Wait a few minutes for the changes to propagate and your records will be created in the hosted zone you chose.

> 5. Repeat the steps as many times as you want, but changing the **Weight** parameter.

Author: ```suggestion
5. Repeat the steps as many times as you want, but changing the **answers_list** and **weight** parameters.
```

> 5. Repeat the steps as many times as you want, but changing the **answers_list** and **weight** parameters.

Author: ```suggestion
5. Repeat the steps to create the other balanced DNS records needed while observing Intelligent DNS limits, and always using the same **entry** and setting the desired addresses in the **answers_list** parameter and weights for each record in the **weight** parameter.
```

>     - Optionally, you can choose different values for the **Name** and **Description** parameters.

Author: ```suggestion
```

:::tip
Check the [Azion API documentation](https://api.azion.com/) and the [OpenAPI specification](https://github.com/aziontech/azionapi-openapi/) to know more about what the Azion API can offer.

> :::

Author: ```suggestion
:::

---

## Testing your balanced record

After configuring your balanced record, you can test and see if it's working via terminal commands:

1. Run `dig +short [your balanced hostname]` a couple of times in a row.
- You'll see the address list vary based on the record selected by the Intelligent DNS load balancer each time.
2. The response will be similar to:

```bash
dig +short lbexample @ns1.aziondns.net
192.111.0.1
```

```bash
dig +short lbexample @ns1.aziondns.net
192.111.0.2
```

```bash
dig +short lbexample @ns1.aziondns.net
192.111.0.1
```
```

---
title: How to perform a load balance between DNS records
description: Perform a DNS load balance by assigning different weights to records configured in Intelligent DNS.
meta_tags: dns, intelligent dns, load balance, records
namespace: docs_secure_load_balance_dns
menu_namespace: secureMenu
permalink: /documentation/products/guides/secure/load-balance-dns/
---

import ContributorList from '~/components/ContributorList.astro'

> You can use a [Intelligent DNS zone](/en/documentation/products/secure/intelligent-dns/) to create more than one record and perform a DNS load balance. Assigning different weights to each record for a same zone helps distribute incoming network traffic, ensuring optimal resource utilization and preventing overload on a single server.

Reviewer: ```suggestion
You can use a [Intelligent DNS zone](/en/documentation/products/secure/intelligent-dns/) to create multiple records and perform a DNS load balance. Assigning different weights to each record for the same zone helps distribute incoming network traffic, ensuring optimal resource utilization and preventing overload on a single server.
```

> You can use a [Intelligent DNS zone](/en/documentation/products/secure/intelligent-dns/) to create more than one record and perform a DNS load balance. Assigning different weights to each record for a same zone helps distribute incoming network traffic, ensuring optimal resource utilization and preventing overload on a single server.

Reviewer 2: ```suggestion
You can use an [Intelligent DNS zone](/en/documentation/products/secure/intelligent-dns/) to create more than one record and perform a DNS load balance. Assigning different weights to each record for a same zone helps distribute incoming network traffic, ensuring optimal resource utilization and preventing overload on a single server.
```

---

## Via RTM

1. [Access RTM](/en/documentation/products/guides/how-to-access-rtm/) > **Intelligent DNS**.
2. From the list, choose the zone in which you want to add records. Or, [create a zone](/en/documentation/products/guides/secure/intelligent-dns-configure-main-settings/#creating-a-new-zone).
3. Select the **Records** tab.
4. Click the **Add Record** button.
5. In **Name**, provide the new record's name. Example: `lbexample`.
6. In **Type**, select the type of record `A`.
7. In **Value**, add the items for the DNS response to the registered record in IPv4 address format. Example: `192.111.0.1`.
8. In **TTL (seconds)**, choose the time, in seconds, that a response can be cached on a resolver server. Maximum value: `2147483647`.
9. In **Policy**, select **Weighted**.
    - In **Weight**, specify the weight for the record, considering that you'll create other records for the same zone with different weights. Accepts values from `0` to `255`.
    - In **Description** (optional), you can add a short text that differentiates the records you'll create for the load balance. Accepts up to *45 characters*.
10. Click the **Save** button.

> 11. Repeat steps 4 to 10 to create the other balanced DNS records needed while observing Intelligent DNS limits, and always using the same **Name** and setting the desired addresses in **Value** and weights for each record in **Weight**.

Reviewer: ```suggestion
11. Repeat steps 4 to 10 to create the other balanced DNS records needed while observing Intelligent DNS limits, always using the same **Name** and setting the desired addresses in **Value** and weights for each record in **Weight**.
```

9. In **Policy**, select **Weighted**.
    - In **Weight**, specify the weight for the record, considering that you'll create other records for the same zone with different weights. Accepts values from `0` to `255`.
    - In **Description** (optional), you can add a short text that differentiates the records you'll create for the load balance. Accepts up to *45 characters*.
10. Click the **Save** button.
11. Repeat steps 4 to 10 to create the other balanced DNS records needed while observing Intelligent DNS limits, always using the same **Name** and setting the desired addresses in **Value** and weights for each record in **Weight**.

If you add 3 records, you can, for example, specify a weight of **50** for the first record, **20** for the second record, and **30** for the third record.

---

## Via API

1. Run the following `GET` request in your terminal, replacing `[TOKEN VALUE]` with your [personal token](/en/documentation/products/guides/personal-tokens/) to retrieve your `<hosted_zone_id>`:

```bash
curl --location 'https://api.azionapi.net/intelligent_dns' \
--header 'Authorization: Token [TOKEN VALUE]' \
--header 'Accept: application/json; version=3'
```

> 2. You'll receive a response with all your existing edge firewalls. Copy the value of the `<id>` that you want to use.

Author: ```suggestion
2. You'll receive a response with all your existing zones. Copy the value of the `<id>` that you want to use.
```

3. Run the following `POST` request, replacing `[TOKEN VALUE]` with your [personal token](/en/documentation/products/guides/personal-tokens/) and the `<hosted_zone_id>` value you copied:

```bash
curl --location 'https://api.azionapi.net/intelligent_dns/<hosted_zone_id>/records' \
--header 'Content-Type: application/json' \
--header 'Authorization: Token [TOKEN VALUE]' \
--header 'Accept: application/json; version=3' \
--data-raw '{
    "record_type": "A",
    "entry": "lbexample",
    "answers_list": [
        "192.111.0.1",
        "192.111.0.2"
    ],
    "ttl": 20,
    "policy": "weighted",
    "weight": 50,
    "description": "policy weight 50"
}'
```

4. You'll receive a response similar to this:

```json
{
  "results": {
    "answers_list": [
      "192.111.0.1",
      "192.111.0.2"
    ],
    "zone_id": 3600,
    "weight": 50,
    "record_type": "A",
    "ttl": 20,
    "policy": "weighted",
    "entry": "lbexample",
    "id": 61752,
    "description": "policy weight 50"
  },
  "schema_version": 3
}
```

Wait a few minutes for the changes to propagate and your records will be created in the hosted zone you chose.

> 5. Repeat the steps to create the other balanced DNS records needed while observing Intelligent DNS limits, and always using the same **entry** and setting the desired addresses in the **answers_list** parameter and weights for each record in the **weight** parameter.

Reviewer 3: ```suggestion
5. Repeat the steps to create the other balanced DNS records needed while observing Intelligent DNS limits, always using the same `entry` and setting the desired addresses in the `answers_list` parameter and weights for each record in the `weight` parameter.
```
